src/main/java/org/jivesoftware/smackx/pubsub/PubSubIntegrationTest.java

```diff
@@ -947,4 +947,50 @@ public void deleteNonExistentNodeTest() throws NoResponseException, XMPPErrorExc
             assertEquals(StanzaError.Condition.item_not_found, e.getStanzaError().getCondition());
         }
     }
+
+    /**
+     * Assert that the server send a notification to subscribers when deleting a
+     * node that exist
+     * 
+     * <p>
+     * From XEP-0060 § 8.4.1:
+     * </p>
+     * <blockquote> In order to delete a node, a node owner MUST send a node
+     * deletion request, consisting of a &lt;delete/&gt; element whose 'node'
+     * attribute specifies the NodeID of the node to be deleted </bloquote>
+     * 
+     * @throws NoResponseException                     if there was no response from
+     *                                                 the remote entity.
+     * @throws XMPPErrorException                      if there was an XMPP error
+     *                                                 returned.
+     * @throws NotConnectedException                   if the XMPP connection is not
+     *                                                 connected.
+     * @throws InterruptedException                    if the calling thread was
+     *                                                 interrupted.
+     * @throws PubSubException.NotAPubSubNodeException if the node cannot be
+     *                                                 accessed.
+     */
+    @SmackIntegrationTest
+    public void deleteNodeAndNotifySubscribersTest() throws NoResponseException, XMPPErrorException,
+            NotConnectedException, InterruptedException, PubSubException.NotAPubSubNodeException {
+        final String nodename = "sinttest-delete-node-that-exist-" + testRunId;
+        final String needle = "<event xmlns='http://jabber.org/protocol/pubsub#event'>";
+        try {
+            LeafNode node = pubSubManagerOne.createNode(nodename);
+            final Node subscriberNode = pubSubManagerTwo.getNode(nodename);
+            final EntityBareJid subscriber = conTwo.getUser().asEntityBareJid();
+            subscriberNode.subscribe(subscriber);
+            final CompletableFuture<Stanza> result = new CompletableFuture<>();
+            conTwo.addAsyncStanzaListener(result::complete, stanza -> stanza.toXML("").toString().contains(needle));
+            // Delete an existent node
+            pubSubManagerOne.deleteNode(nodename);
+            assertNull(result.get(conOne.getReplyTimeout(), TimeUnit.MILLISECONDS));
```

Comment:
At this point (line 987) do you expect there to be a result?

`assertNull` means: "I expect this to be null. Fail the test if it is not null"
`assertNotNull` means: "I expect this to be not null. Fail the test if it is null"

Reply:
I will use assertNotNull